docs/io.cozy.files.md

```diff
@@ -284,3 +284,9 @@ fields), and a relationship to the versioned file.
   }
 }
 ```
+
+## `io.cozy.files.settings`
+
+This doctype is used to store settings for files. It is currenly only used for the qualification migration service.
+
+- lastProcessedFileDate: {string} the date of the last processed file. It is used to know from where the query should be starting on an index sorted by date.
```

Comment:
Which date? For files, we have `created_at`, `updated_at`, `cozyMetadata.createdAt`, `cozyMetadata.updatedAt`, and `cozyMetadata.uploadAt`.

And why use a date instead of a sequence number from the changes feed? It would be more reliable to process old files that are imported by a sharing.

Reply:
It's `cozyMetadata.updatedAt`, I'll clarify that. A date is useful to query an index, although it's also possible to use the change feed. 
Why would it be more reliable for files imported by a sharing?

Reply:
When a file is imported by a sharing, its `cozyMetadata.updatedAt` can be in the past, and it's possible that there are a lot of files with a greater date. So, if the file is imported while the migration service is running, it can be missed.

Reply:
Erf ok, I thought this attribute was updated in this case :/
If it's a synced sharing, the file will be treated on the sharer side and eventually propagated anyway, but there might be a glitch if it's a "one-shot" sharing and the service didn't run on the sharer side before the sharing happened..
As there are several places in Cozy where we need to process files based on a date / seq changes, I think it would be worth it to document some kind of good practices and behaviour like this that should be kept in mind

Reply:
Mentioned in https://github.com/cozy/cozy-drive/issues/2174